enterprise/cluster/src/main/java/org/neo4j/cluster/protocol/atomicbroadcast/multipaxos/LearnerState.java

```diff
@@ -252,7 +254,7 @@ public LearnerState handle( LearnerContext context,
 
                                 context.setLastKnownLearnedInstanceInCluster(
                                         catchUpTo,
-                                        context.getIdForUri( new URI( message.getHeader( Message.FROM ) ) ) );
+                                        new org.neo4j.cluster.InstanceId( Integer.parseInt( message.getHeader( Message.INSTANCE_ID ) )));
```

Comment:
This will cause issues if we run mixed version clusters - the old version instances will not contain the INSTANCE_ID header and will cause failures. We should probably have this as an if check that falls back to URI based lookups as before.